sklearn/decomposition/tests/test_pca.py

```diff
@@ -686,3 +689,18 @@ def test_variance_correctness(copy):
     pca_var = pca.explained_variance_ / pca.explained_variance_ratio_
     true_var = np.var(X, ddof=1, axis=0).sum()
     np.testing.assert_allclose(pca_var, true_var)
+
+
+def test_array_api_compliance():
+    # TODO: rewrite me as a pytest parametrized test with a parametrized
+    # generator instead of a for loop
+    estimators = [
+        PCA(n_components=2, svd_solver="full"),
+        # PCA(n_components=2, svd_solver="randomized", power_iteration_normalizer="QR"),
```

Comment:
This is still not tested and would fail if uncommented. Maybe we could focus this PR on the "full" solver and handle the randomized svd solver in a follow-up PR because this PR is already quite big.